Sources/SwifterSwift/UIKit/UIButtonExtensions.swift

```diff
@@ -179,6 +179,22 @@ public extension UIButton {
             contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
         }
     }
+    
+    /// SwifterSwift: Set background color for specified state.
+    /// - Parameters:
+    ///   - color: UIColor.
+    ///   - forState: set the UIControl.State for the desired color.
+    func setBackgroundColor(color: UIColor, forState: UIControl.State) {
+        clipsToBounds = true  // maintain corner radius
+        UIGraphicsBeginImageContext(CGSize(width: 1, height: 1))
```

Comment:
Should we use `UIGraphicsImageRenderer` APIs here? I remember a while ago that this is the recomended way to do it. See https://github.com/SwifterSwift/SwifterSwift/pull/721/files for example. I guess (not sure) we have iOS 10 as minimum for a while so conditional checking ...

Reply:
Thanks. That really makes sense. I changed using `UIGraphicsImageRenderer` for `iOS 10.0` and `tvOS 10.0`. It worked fine and tests passed.

```swift
    /// SwifterSwift: Set background color for specified state.
    /// - Parameters:
    ///   - color: UIColor.
    ///   - forState: set the UIControl.State for the desired color.
    func setBackgroundColor(color: UIColor, forState: UIControl.State) {
        clipsToBounds = true  // maintain corner radius
        
        if #available(iOS 10.0, tvOS 10.0, *) {
            let colorImage = UIGraphicsImageRenderer(size: CGSize(width: 1, height: 1)).image {
                context in
                color.setFill()
                context.fill(CGRect(x: 0, y: 0, width: 1, height: 1))
                draw(.zero)
            }
            setBackgroundImage(colorImage, for: forState)
            return
        }
        
        UIGraphicsBeginImageContext(CGSize(width: 1, height: 1))
        if let context = UIGraphicsGetCurrentContext() {
            context.setFillColor(color.cgColor)
            context.fill(CGRect(x: 0, y: 0, width: 1, height: 1))
            let colorImage = UIGraphicsGetImageFromCurrentImageContext()
            UIGraphicsEndImageContext()
            setBackgroundImage(colorImage, for: forState)
        }
    }
```

Reply:
Yes for sure! 
Do we need the if #available? I think we have minimums as 10 already

Reply:
I guess we might need to check with #available. Even the PR you provided earlier checked with it [here](https://github.com/SwifterSwift/SwifterSwift/pull/721/files). Shall we revert it back?

> 1 Error
> 🚫	**[Sources/SwifterSwift/UIKit/UIButtonExtensions.swift#L190](https://github.com/TwizzyIndy/SwifterSwift/blob/a16139b5efcb000a524817afcb9c86f14a3b24d8/Sources/SwifterSwift/UIKit/UIButtonExtensions.swift#L190)**: 'UIGraphicsImageRenderer' is only available in tvOS 10.0 or newer
> ` let colorImage = UIGraphicsImageRenderer(size: CGSize(width: 1, height: 1)).image { context in `
> 1 Message
> 📖	Thank you for submitting a pull request to SwifterSwift. The team will review your submission as soon as possible.
> Generated by 🚫 [Danger](https://danger.systems/)

Reply:
Yeah, sorry for that... I was pretty sure we had this as mim